File: ios/Plugin/M1CheckerPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(M1CheckerPlugin, "M1Checker",
    CAP_PLUGIN_METHOD(runningOnM1, CAPPluginReturnPromise);
)

File: ios/Plugin/M1CheckerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(M1CheckerPlugin)
public class M1CheckerPlugin: CAPPlugin {
    private let implementation = M1Checker()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
